types/validator_set.go

```diff
@@ -986,17 +986,21 @@ func ValidatorSetFromProto(vp *tmproto.ValidatorSet) (*ValidatorSet, error) {
 
 // ValidatorSetFromExistingValidators takes an existing array of validators and rebuilds
 // the exact same validator set that corresponds to it without changing the proposer priority or power
-func ValidatorSetFromExistingValidators(valz []*Validator) (*ValidatorSet, error) {
-	if len(valz) == 0 {
-		return nil, errors.New("no validators given")
+// if any of the validators fail validate basic then an empty set is returned.
+func ValidatorSetFromExistingValidators(valz []*Validator) *ValidatorSet {
+	for _, val := range valz {
+		err := val.ValidateBasic()
+		if err != nil {
+			return &ValidatorSet{}
```

Comment:
We shouldn't just silently discard errors, it makes debugging much harder.

Reply:
should I panic. Is it a developer fault if some / all of the validators are empty? or should I return an error?

Reply:
Return an error. The input to this function can be an external source.

Reply:
ok